pyccel/ast/variable.py

```diff
@@ -212,6 +217,10 @@ def __init__(
         if self._rank < 2:
             self._order = None
 
+        if self.is_ndarray and not self.is_tuple\
```

Comment:
Are there cases where `self.is_ndarray` and `self.is_tuple`? If so this seems like a bug that needs to be fixed